src/tools/tc_utils/tc_diag/tc_diag_conf_info.cc

```diff
@@ -0,0 +1,452 @@
+// *=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*
+// ** Copyright UCAR (c) 1992 - 2023
+// ** University Corporation for Atmospheric Research (UCAR)
+// ** National Center for Atmospheric Research (NCAR)
+// ** Research Applications Lab (RAL)
+// ** P.O.Box 3000, Boulder, Colorado, 80307-3000, USA
+// *=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*
+
+////////////////////////////////////////////////////////////////////////
+
+using namespace std;
+
+#include <dirent.h>
+#include <iostream>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <cmath>
+
+#include "tc_diag_conf_info.h"
+
+#include "vx_log.h"
+
+////////////////////////////////////////////////////////////////////////
+//
+//  Code for struct DataOptInfo
+//
+////////////////////////////////////////////////////////////////////////
+
+void DataOptInfo::clear() {
+
+   tech_ids.clear();
+   data_files.clear();
+
+   return;
+}
+
+////////////////////////////////////////////////////////////////////////
+
+DataOptInfo & DataOptInfo::operator+=(const DataOptInfo &info) {
+
+   tech_ids.add(info.tech_ids);
+   data_files.add(info.data_files);
+
+   return(*this);
+}
+
+////////////////////////////////////////////////////////////////////////
+//
+//  Code for class DomainInfo
+//
+////////////////////////////////////////////////////////////////////////
+
+DomainInfo::DomainInfo() {
+
+   init_from_scratch();
+}
+
+////////////////////////////////////////////////////////////////////////
+
+DomainInfo::~DomainInfo() {
+
+   clear();
+}
+
+////////////////////////////////////////////////////////////////////////
+
+void DomainInfo::init_from_scratch() {
+
+   clear();
+
+   return;
+}
+
+////////////////////////////////////////////////////////////////////////
+
+void DomainInfo::clear() {
+
+   tech_ids.clear();
+   data_files.clear();
+   domain.clear();
+
+   data.name         = (const char *) 0;
+   data.range_n      = bad_data_int;
+   data.azimuth_n    = bad_data_int;
+   data.range_max_km = bad_data_double;
+   data.lat_center   = bad_data_double;
+   data.lon_center   = bad_data_double;
+
+   delta_range_km = bad_data_double;
+
+   var_info_ptr.clear();
+   diag_script.clear();
+
+   return;
+}
+
+////////////////////////////////////////////////////////////////////////
+
+void DomainInfo::parse_domain_info(Dictionary &dict) {
+
+   // Initialize
+   clear();
+
+   // Note: tech_ids and data_files are specified on the
+   //       command line rather than in the config file
+
+   // Conf: domain
+   domain = dict.lookup_string(conf_key_domain);
+
+   // Hard-code the name
+   data.name = "TCRMW";
```

Comment:
Should this be TCDIAG? Or is the data still considered "TCRMW"?

Reply:
I originally thought that hard-coded string was necessary, but I switched it to `TCDIAG` with this [commit](https://github.com/dtcenter/MET/commit/b6ee12a6c111928772c055105f96b823d1341a20).